ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -214,7 +226,7 @@ public void run() {
               logger.trace("{} being checked for compaction", store);
               boolean compactionStarted = false;
               try {
-                if (store.isStarted() && !storesToSkip.contains(store)) {
+                if (store.isStarted() && !storesToSkip.contains(store) && !storesDisabledCompaction.contains(store)) {
```

Comment:
this part of the code needs to be tested I think - the fact that compaction won't start on a store if it has been disabled.

Reply:
New test has been added.